tutorials/haskell/hspec-golden-test-tutorial/tutorial.md

```diff
@@ -0,0 +1,506 @@
+fizzBuzz list = map fizzOrBuzz list
+
+fizzOrBuzz ::Int -> String
+fizzOrBuzz n | n `mod` 15 == 0  = "FizzBuzz"
+             | n `mod` 3  == 0  = "Fizz"
+             | n `mod` 5  == 0  = "Buzz"
+             | otherwise        = show n
+
+```
+For simplicity we will test a simple function `fizzbuzz` that replaces multiples of 3 with Fizz and multiples of 5 with Buzz. In case that a number is a multiple of both it replaces the number with FizzBuzz. Before moving on with testing, let's review some of `hspec-golden` [documentation](http://hackage.haskell.org/package/hspec-golden-0.1.0.1/docs/Test-Hspec-Golden.html).
+
+Hspec-golden provides us with a `defaultGolden` function which creates the golden files and compares it with the Subject Under Test (SUT) output. It takes two parameters:
+
+  - the **name** of the test 
+  - the output of our SUT 
+
+**Note** : _It is important to give the tests unique names otherwise `hspec-golden` won't work_.
+
+```haskell
+defaultGolden :: String -> String -> Golden String
+```
+
+By default this function searches for golden tests inside a `.golden/` directory. This directory is created automatically when the test runs.
+
+Okay, enough reading, let's create our tests. Let's first create a test module named `FizzBuzzGoldenSpec` under `test/FizzBuzz/` directory.
+
+```haskell
+module FizzBuzz.FizzBuzzGoldenSpec where
+```
+
+We need some imports.
+
+```haskell
+import           Test.Hspec                   --Tests
+import           Test.Hspec.Golden            --Golden Tests
+import           FIZZBUZ.FizzBuzz             -- SUT
+```
+Finally let's write the test.
+
+```haskell
+spec :: Spec
+spec = 
+    describe "fizzBuzz" $
+      it "Turns 3 multiples to fizz and 5 multiples to buzz" $
+        defaultGolden "fizzbuzz" (show $ fizzBuzz [1,2,3,4,5,11,12,13,14,15])
+```
+We are now ready to test our module.
+
+```shell
+$ stack test
+
+hspec-golden-tests> test (suite: hspec-golden-tests-test)
+
+
+FizzBuzz.FizzBuzzGolden
+  fizzBuzz
+    Turns 3 multiples to fizz and 5 multiples to buzz
+      First time execution. Golden file created.
+
+Hello.Hello
+  sayHi
+    shows a Hello Golden Testers string
+
+Finished in 0.0006 seconds
+2 examples, 0 failures
+
+hspec-golden-tests> Test suite hspec-golden-tests-test passed
+```
+
+The test is successful. If we check the ".golden" directory we will find two files.
+
+ - golden : Stores the expected output.
+ - actual : Stores the real output from the SUT.
+
+
+
+```shell
+$ tree
+.golden
+ └── fizzbuzz
+     ├── actual
+     └── golden
+```
+
+The testing framework recognized that this was the first execution, therefore created the fizzbuzz test with the `actual` and `golden` files. The difference between these two files is that the `golden` file will stay the same unless we want to update it while the `actual` file will be overwritten every time we run the test. This is useful when updating the tests, but we will see that later on in this tutorial. 
```

Comment:
```suggestion
The testing framework recognized that this was the first execution, therefore it created the fizzbuzz test with the `actual` and `golden` files. The difference between these two files is that the `golden` file will stay the same unless we want to update it while the `actual` file will be overwritten every time we run the test. This is useful when updating the tests, but we will see that later in this tutorial. 
```

Reply:
If you say "the `golden` file will stay the same while the `actual` file will be overwritten every time we run the test" I understand that perfectly. But when you add more information, like "unless we want to update it", in the middle of the sentence, it kinda lost me. Maybe you can add that piece of information in a different sentence?